src/coreclr/utilcode/sstring.cpp

```diff
@@ -1939,6 +1795,7 @@ void SString::VPrintf(const CHAR *format, va_list args)
     CONTRACT_VOID
     {
         INSTANCE_CHECK;
+        PRECONDITION(GetRepresentation() == REPRESENTATION_ASCII || GetRepresentation() == REPRESENTATION_UTF8);
```

Comment:
Why do we care about the representation of the existing content? The existing content will be overwritten so it should not matter.

Reply:
For some reason I got confused between the `AppendPrintf` method and this one and didn't realize that this one overrides all of the contents.